### ISSUE TITLE
Database may be already in use

### ISSUE THREAD

**Issue by ENMAGames** [EsotericSoftware/clippy#3]
On one of my machines, I can't seem to start clippy anymore. It logs the following exception.
I'm gonna try deleting the .clippy folder, which might work, but in any case:

00:00 ERROR: Error opening clip database.
org.h2.jdbc.JdbcSQLException: Database may be already in use: "Locked by another computer: 169.254.80.204". Possible solutions: close all other connection(s); use the server mode [90020-181]
	at org.h2.message.DbException.getJdbcSQLException(DbException.java:345)
	at org.h2.message.DbException.get(DbException.java:179)
	at org.h2.message.DbException.get(DbException.java:155)
	at org.h2.store.FileLock.getExceptionAlreadyInUse(FileLock.java:461)
	at org.h2.store.FileLock.lockSocket(FileLock.java:397)
	at org.h2.store.FileLock.lock(FileLock.java:138)
	at org.h2.engine.Database.open(Database.java:631)
	at org.h2.engine.Database.openDatabase(Database.java:266)
	at org.h2.engine.Database.<init>(Database.java:260)
	at org.h2.engine.Engine.openSession(Engine.java:60)
	at org.h2.engine.Engine.openSession(Engine.java:167)
	at org.h2.engine.Engine.createSessionAndValidate(Engine.java:145)
	at org.h2.engine.Engine.createSession(Engine.java:128)
	at org.h2.engine.Engine.createSession(Engine.java:26)
	at org.h2.engine.SessionRemote.connectEmbeddedOrServer(SessionRemote.java:347)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:108)
	at org.h2.jdbc.JdbcConnection.<init>(JdbcConnection.java:92)
	at org.h2.Driver.connect(Driver.java:72)
	at java.sql.DriverManager.getConnection(Unknown Source)
	at java.sql.DriverManager.getConnection(Unknown Source)
	at com.esotericsoftware.clippy.util.DataStore.openConnection(DataStore.java:161)
	at com.esotericsoftware.clippy.util.DataStore.open(DataStore.java:116)
	at com.esotericsoftware.clippy.ClipDataStore.<init>(ClipDataStore.java:41)
	at com.esotericsoftware.clippy.Clippy.<init>(Clippy.java:84)
	at com.esotericsoftware.clippy.Clippy$5.run(Clippy.java:186)
	at java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$500(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.run(Unknown Source)

**Comment by ENMAGames** [comment]
Deleting the .clippy folder did work, fyi.

**Comment by NathanSweet** [comment]
Sounds like Clippy was already running? Maybe it crashed in a way that didn't terminate the process but left the DB open.

**Comment by ENMAGames** [comment]
It does sound like that, but I triple checked to see if I see it in the processes and nothing. Ans this happened on this particular machine even after reboot.
Although I looked for "clippy.exe" not java.exe, I might have missed that one.

The only time I made clippy crash in general, that I know of, was copying a gigantic piece of text, probably overflowing its memory... But it did run after that, just without that entry in the DB.
